### ISSUE TITLE
some errors

### ISSUE THREAD

**Issue by jackeykong** [jamesdbrock/hffix#43]
1. On Windows, there is no 'ssize_t' type. 
2. the 'std::string_view' has no begin and end members.
   ` void push_back_string(int tag, std::string_view s) {
        push_back_string(tag, s.begin(), s.end());
    }`

**Comment by jamesdbrock** [comment]
Hi, thanks for opening this issue.

> 1. On Windows, there is no 'ssize_t' type.

You're right, `ssize_t` is not standard and we shouldn't use it. I'll fix that.

> 2. the 'std::string_view' has no begin and end members.
>    ` void push_back_string(int tag, std::string_view s) { push_back_string(tag, s.begin(), s.end()); }`

`std::string_view` `begin` and `end` are standard, starting in C++17. https://en.cppreference.com/w/cpp/string/basic_string_view Do you have a suggestion for what to do about that?

**Comment by jamesdbrock** [comment]
I tagged v1.1.1 for the `ssize_t` bugfix. https://github.com/jamesdbrock/hffix/releases/tag/v1.1.1

**Comment by jackeykong** [comment]
> Hi, thanks for opening this issue.
> 
> > 1. On Windows, there is no 'ssize_t' type.
> 
> You're right, `ssize_t` is not standard and we shouldn't use it. I'll fix that.
> 
> > 1. the 'std::string_view' has no begin and end members.
> >    ` void push_back_string(int tag, std::string_view s) { push_back_string(tag, s.begin(), s.end()); }`
> 
> `std::string_view` `begin` and `end` are standard, starting in C++17. https://en.cppreference.com/w/cpp/string/basic_string_view Do you have a suggestion for what to do about that?

**Comment by jackeykong** [comment]
I also found a small error in the Multi-threaded Sending section of the readme.md file.
`hffix::message_reader::const_iterator i = std::find(r.begin(), r.end(), hffix::tag_equal(hffix::tag::MsgSeqNum)); 
if (i != r.end()) 
{
    std::snprintf(const_cast<char*>(i->begin()), i->size(), "%.8i", next_sequence_number++); 
}`

1. should be use `std::find_if` to find out iterator i
2. iterator i has no begin and size members

**Comment by jackeykong** [comment]
> Hi, thanks for opening this issue.
> 
> > 1. On Windows, there is no 'ssize_t' type.
> 
> You're right, `ssize_t` is not standard and we shouldn't use it. I'll fix that.
> 
> > 1. the 'std::string_view' has no begin and end members.
> >    ` void push_back_string(int tag, std::string_view s) { push_back_string(tag, s.begin(), s.end()); }`
> 
> `std::string_view` `begin` and `end` are standard, starting in C++17. https://en.cppreference.com/w/cpp/string/basic_string_view Do you have a suggestion for what to do about that?

But, `std::string_view::begin` and `std::string_view::end` return `std::basic_string_view<CharT,Traits>` instead of `const char*`

**Comment by jackeykong** [comment]
> > But, `std::string_view::begin` and `std::string_view::end` return `std::basic_string_view<CharT,Traits>` instead of `const char*`
> 
> They return a `std::basic_string_view<char>::const_iterator` which will be type `const char *`. Or is it not type `const char *` in your Windows environment? I guess if it's not `const char *` then `push_back_string(int tag, std::string_view s)` will fail to typecheck, is that what's happening for you?
> 
> https://en.cppreference.com/w/cpp/string/basic_string_view
> 
> https://docs.microsoft.com/en-us/cpp/standard-library/string-view-typedefs?view=msvc-160#string_view

Yes, i use visual studio 2017 on windows, the error message is `error C2664: 'void hffix::message_writer::push_back_string(int,std::string_view)': cannot convert argument 2 from 'std::_String_view_iterator<_Traits>' to 'const char *'`

**Comment by jamesdbrock** [comment]
I don't have easy access to a Windows machine so it's inconvenient for me to write a fix for this because I can't test it on Windows. Would you like to make a PR to fix this problem?

**Comment by enricodetoma** [comment]
I created pull request #56 after encountering the same problem with MSVC
